hydromt/_validators/data_catalog.py

```diff
@@ -46,21 +49,31 @@ class Extent(BaseModel):
 class DataCatalogMetaData(BaseModel):
     """The metadata section of a Hydromt data catalog."""
 
-    root: Optional[Path] = None
+    roots: List[Path]
     version: Optional[Union[str, Number]] = None
+    hydromt_version: str
```

Comment:
I see in from_yml and from_dict that version can still be None. So should the hydromt_version here also be optional?

Reply:
That version refers to the version of the catalogs, which it may or may not have, but `hydromt_version` is the version of hydromt that the catalog is compatible with. I think it is good to force people to specify what versions they are compatible with, since if they don't they won't get warnings if we make incompatible changes and use their old catalogs. What do you think?

Reply:
It's good to encourage it but not sure we want to force it. Especially since after v1 we hope the syntax and keywords of the catalog should be more stable so this might be a lot of trouble for the user to find out this information and the syntax and keep it updated with new hydromt versions. 
Maybe if None just print a warning that if the hydromt_version was not found, errors can happen later due to that. And do add the current used version? Then if the catalog is exported later it will print and can be re-used?
@DirkEilander what do you think?

Reply:
I'm okay with this solution, I'll go with that